src/Traits/ReflectionClassLikeTrait.php

```diff
@@ -442,7 +443,7 @@ public function getParentClass()
             $extendsNode = $hasExtends ? $this->classLikeNode->$extendsField : null;
             if ($extendsNode instanceof FullyQualified) {
                 $extendsName = $extendsNode->toString();
-                $parentClass = class_exists($extendsName, false) ? new parent($extendsName) : new static($extendsName);
+                $parentClass = class_exists($extendsName, false) ? new BaseReflectionClass($extendsName) : new ReflectionClass($extendsName);
```

Comment:
Fact, that trait is used only once and seems like it would be for a long time (unless PHP decides to reflect class/interface/trait using different reflection classes) is a problem that I think needs to be solved.

If we inline trait (copy trait code into the only class using it and drop trait), then problem should go away by itself. This is a major change however (even though not a BC break) and likely a @lisachenko approval is required before making such change.

Actually better to move it out into PR of it's own.